force-app/main/classes/FFPoller.cls

```diff
@@ -0,0 +1,209 @@
+        return this;
+    }
+
+    public void execute() {
+        execute(null);
+    }
+
+    public void execute(SchedulableContext schedulableContext) {
+        System.enqueueJob(new FFPollAsync(configuration, iteration));
+    }
+
+
+    // INNER
+
+    public class Configuration {
+        private Callable pollInstance { get; private set; }
+        private Callable checkCompletionInstance { get; private set; }
+        private Callable callbackInstance { get; private set; }
+        private List<IncrementalDelay> incremDelays { get; private set; }
+        private Datetime startTime { get; private set; }
+        private Integer timeout { get; private set; }
+
+        public Configuration() {
+            this.startTime = Datetime.now();
+            this.timeout = 0;
+        }
+    }
+
+    private class IncrementalDelay implements Comparable {
+        private Integer delayInSeconds;
+        private Integer upperIteration;
+
+        public IncrementalDelay(Integer upperIteration, Integer delayInSeconds) {
+            this.delayInSeconds = delayInSeconds;
+            this.upperIteration = upperIteration;
+        }
+
+        public Integer compareTo(Object compareTo) {
+            IncrementalDelay inc = (IncrementalDelay) compareTo;
+            return upperIteration > inc.upperIteration ? -1 : (upperIteration < inc.upperIteration) ? -1 : 0;
+        }
+    }
+
+    @TestVisible
+    private class FFPollAsync implements Queueable, Database.AllowsCallouts {
+        private Configuration configuration;
+        private Integer iteration;
+    
+        // CONSTRUCTOR
+    
+        public FFPollAsync(Configuration configuration, Integer iteration) {
+            this.configuration = configuration;
+            this.iteration = iteration;
+        }
+    
+        // PUBLIC
+        public void execute(QueueableContext context) {
+            FFPollAsyncFinalizer finalizer = new FFPollAsyncFinalizer(this);
+            System.attachFinalizer(finalizer);
+    
+            Object pollResponse = invokeClass(configuration.pollInstance, null);
+            Boolean completed = (Boolean) invokeClass(configuration.checkCompletionInstance, pollResponse);
+    
+            if (completed) {
+                invokeClass(configuration.callbackInstance, pollResponse);
+            } else {
+                scheduleAgain();
+            }
+        }
+    
+        // PRIVATE
+    
+        private Object invokeClass(Callable instance, Object args) {
+            return instance.call('', new Map<String, Object>{ 'default' => args });
+        }
+
+        private void scheduleAgain() {
+            try {
+                String nextFireCron = calculateNextFireCron(iteration, configuration);
+    
+                iteration++;
+                FFPoller scheduledPoll = new FFPoller(configuration, iteration);
+                System.schedule('Polling - ' + nextFireCron, nextFireCron, scheduledPoll);
```

Comment:
Can we parameterize job name. 
In our use case, We have to find if Poller is already running for a tenant, do not start it again, and if for some reason it get killed, then start this again from different schedule job.